src/sinks/postgres/config.rs

```diff
@@ -0,0 +1,144 @@
+use futures::FutureExt;
+use tower::ServiceBuilder;
+use vector_lib::{
+    config::AcknowledgementsConfig,
+    configurable::{component::GenerateConfig, configurable_component},
+    sink::VectorSink,
+};
+
+use super::{
+    service::{PostgresRetryLogic, PostgresService},
+    sink::PostgresSink,
+};
+use sqlx::{postgres::PgPoolOptions, Pool, Postgres};
+
+use crate::{
+    config::{Input, SinkConfig, SinkContext},
+    sinks::{
+        util::{
+            BatchConfig, RealtimeSizeBasedDefaultBatchSettings, ServiceBuilderExt,
+            TowerRequestConfig, UriSerde,
+        },
+        Healthcheck,
+    },
+};
+
+const fn default_pool_size() -> u32 {
+    5
+}
+
+/// Configuration for the `postgres` sink.
+#[configurable_component(sink("postgres", "Deliver log data to a PostgreSQL database."))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct PostgresConfig {
+    // TODO: if I used UriSerde instead of String, I couldn't get a url string to use
+    // in the connection pool, as the password would be redacted with UriSerde::to_string
+    /// The connection string for the PostgreSQL server. It can contain the username and password.
+    pub endpoint: String,
+
+    /// The table that data is inserted into.
+    pub table: String,
+
+    /// The postgres connection pool size.
```

Comment:
Here it would be useful to explain what this pool is used for. Maybe a link to relevant docs would suffice.

Reply:
done in 21f3ad3a56090b94cb8f58ec7bf3bfbf4fb51439. Do you think it is enough?

I also have doubts about using a connection pool. Can the event batches be executed in parallel for this sink? I don't know the specifics of vector's internals... https://github.com/vectordotdev/vector/blob/21f3ad3a56090b94cb8f58ec7bf3bfbf4fb51439/src/sinks/postgres/sink.rs#L24

If the batches of events can be processed in parallel, then a connection pool is beneficial. If the batches are processed sequentially, then we should use a single postgres connection as a pooled connection does not have sense